config/config.go

```diff
@@ -56,11 +70,85 @@ func (c *Config) Validate() error {
 	return nil
 }
 
+const (
+	remoteSection = "remote"
+	coreSection   = "core"
+	fetchKey      = "fetch"
+	urlKey        = "url"
+	bareKey       = "bare"
+)
+
+// Unmarshal parses a git-config file and stores it
+func (c *Config) Unmarshal(b []byte) error {
```

Comment:
the main reason is because I tried to mimic the behavior if the Marshal/Unmarshal in other packages, since the size of the config is small, and we are marshaling the content to a content, I found this interface more familiar than others.